packages/api/src/external/hie/set-doc-query-progress-with-source.ts

```diff
@@ -108,18 +73,79 @@ export async function appendDocQueryProgressWithSource({
     return updatedPatient;
   });
 
-  await processDocQueryProgressWebhook({
+  await handleWebhookBeingSent({
     patient: result.dataValues,
-    documentQueryProgress: result.data.documentQueryProgress,
     requestId,
   });
 
   return result;
 }
 
+export function aggregateAndSetHieProgresses(
+  existingPatient: PatientModel,
+  updatedExternalData: PatientExternalData,
+  requestId: string
+) {
+  const documentQueryProgress = !existingPatient.data.documentQueryProgress
+    ? {}
+    : existingPatient.data.documentQueryProgress;
+
+  existingPatient.data.externalData = updatedExternalData;
```

Comment:
I would keep this outside of this function. IMO this function should do one thing only: update the consolidated DQ progress based on a source's DQ progress.